src/commands/trace/__tests__/trace.test.ts

```diff
@@ -0,0 +1,54 @@
+// tslint:disable: no-string-literal no-null-keyword
+import {TraceCommand} from '../trace'
+
+describe('trace', () => {
+  describe('signalToNumber', () => {
+    test('should map undefined to undefined', () => {
+      process.env = {}
+      const write = jest.fn()
+      const command = new TraceCommand()
+      command.context = {stdout: {write}} as any
+
+      expect(command['signalToNumber'].call(undefined, null)).toBeUndefined()
+    })
+    test('should map SIGKILL to 137', () => {
+      process.env = {}
+      const write = jest.fn()
+      const command = new TraceCommand()
+      command.context = {stdout: {write}} as any
+
+      expect(command['signalToNumber'].call(undefined, 'SIGKILL')).toEqual(137)
+    })
+  })
+
+  describe('getData', () => {
+    test('should throw if no CI is detected', () => {
+      process.env = {}
+      const write = jest.fn()
+      const command = new TraceCommand()
+      command.context = {stdout: {write}} as any
+
+      expect(command['getData'].bind(command)).toThrow(
+        'Cannot detect any CI Provider. This command only works if run as part of your CI.'
```

Comment:
is this what we want? This would break under an unsupported CI provider, which might not be a great user experience. It might make more sense to just run the command as is and log a warning?

Reply:
Not sure about this one. I understand your point, but at the same time I think an explicit error is easier to debug during onboarding. So, if someone is trying out custom spans for the first time and we just log a warning, they might overlook it and open up a support ticket saying "why aren't custom spans working?".

However, I think there's 2 scenarios where having a warning is better than failing:
1. If we remove support for a specific CI provider in datadog-ci: the user experience here would be that they upgrade the CLI and it suddenly fails. Not a great UX. However, I don't think this is a likely scenario.
2. If the user is migrating from one CI provider to another one, and they have a period of time where they are running the jobs in both CI providers, and one is supported and the other one isn't. Again, not a super-likely scenario, but more likely that the first one.

How about if I keep the failing behavior by default and add a `--no-fail` param that people can opt into if they are migration CI Providers?

Reply:
that sounds good to me. You can add the `--no-fail` at a later point 👍